java/vector/src/main/java/org/apache/arrow/vector/ipc/JsonFileReader.java

```diff
@@ -329,6 +329,36 @@ protected ArrowBuf read(BufferAllocator allocator, int count) throws IOException
       }
     };
 
+    BufferReader FIXEDSIZEBINARY = new BufferReader() {
+      @Override
+      protected ArrowBuf read(BufferAllocator allocator, int count) throws IOException {
+        ArrayList<byte[]> values = Lists.newArrayList();
+        int byteWidth = 0;
+        for (int i = 0; i < count; i++) {
+          parser.nextToken();
+          final byte[] value = decodeHexSafe(parser.readValueAs(String.class));
+          values.add(value);
+          if (value.length > 0) {
+            if (byteWidth == 0) {
+              byteWidth = value.length;
+            } else if (byteWidth != value.length) {
+              throw new IOException("mismatch byte width (" + value.length + ") at index " + i + ", expecting " + byteWidth);
```

Comment:
Does this mean the input is malformatted? If this is, then it probably shouldn't be IOException..

Reply:
Changed to RuntimeException